src/StreamJsonRpc/JsonMessageFormatter.cs

```diff
@@ -388,6 +455,42 @@ private JsonRpcRequest ReadRequest(JToken json)
                 args is JArray argsArray ? (object)PartiallyParsePositionalArguments(argsArray) :
                 null;
 
+            // If method is $/progress, get the progress instance from the dictionary and call Report
+            string method = json.Value<string>("method");
+
+            if (string.Equals(method, ProgressRequestSpecialMethod, StringComparison.Ordinal))
+            {
+                try
+                {
+                    JToken progressId =
+                        args is JObject ? args["token"] :
+                        args is JArray ? args[0] :
+                        null;
+
+                    JToken value =
+                        args is JObject ? args["value"] :
+                        args is JArray ? args[1] :
+                        null;
+
+                    lock (this.progressMap)
```

Comment:
>progressMap [](start = 31, length = 11)

I'm always weary of locking on the same object you're trying to access within the lock.  I would propose using the SyncRoot property on IDictionary, or just creating a lock object itself: [https://stackoverflow.com/questions/29557718/correct-way-to-lock-the-dictionary-object](https://stackoverflow.com/questions/29557718/correct-way-to-lock-the-dictionary-object)